File: Archived2/Java/Silver/19621.java
import java.io.*;
import java.util.*;

public class Main {

    private static final BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

    public static void main(String[] args) throws IOException {

        int N = Integer.parseInt(br.readLine());
        ArrayList<Meeting> meetings = new ArrayList<>(N);

        for (int i=0;i<N;i++) {
            int[] input = Arrays.stream(br.readLine().split(" ")).mapToInt(Integer::parseInt).toArray();
            meetings.add( new Meeting( input[0], input[1], input[2] ));
        }

        meetings.sort(Comparator.comparingInt(meeting -> meeting.startTime));
        int[] maxUserPerMeeting = meetings.stream().mapToInt((meeting) -> meeting.users).toArray();

        for (int i=0;i<N;i++) {
            int maxUserBefore = 0;
            for (int j=0;j<i;j++) {
                if (meetings.get(i).startTime >= meetings.get(j).endTime) {
                    if (maxUserBefore < maxUserPerMeeting[j]) maxUserBefore = maxUserPerMeeting[j];
                }

            }
            maxUserPerMeeting[i] += maxUserBefore;
        }
        System.out.println(Arrays.stream(maxUserPerMeeting).max().getAsInt());
    }

}

class Meeting {
    int startTime;
    int endTime;
    int users;

    Meeting(int start, int end, int users) {
        this.startTime = start;
        this.endTime = end;
        this.users = users;
    }

}

File: Archived2/Kotlin/Silver/19621.kt
import java.io.BufferedReader
import java.io.InputStreamReader


fun main() {
    val br = BufferedReader(InputStreamReader(System.`in`))
    val n = br.readLine().toInt()
    val meetings = arrayListOf<Meeting>()

    repeat(n) {
        val meetingData = br.readLine().split(" ").map { it.toInt() }
        meetings.add(Meeting(
            startTime = meetingData[0],
            endTime = meetingData[1],
            users = meetingData[2]
        ))
    }

    meetings.sortWith( compareBy { it.startTime } )
    val maxUserPerMeeting = meetings.map { it.users }.toMutableList()
    for (i in 0 until n) {
        var maxUserCount = 0
        for (j in 0 until i) {
            if (meetings[i].startTime >= meetings[j].endTime) {
                if (maxUserCount < maxUserPerMeeting[j]) maxUserCount = maxUserPerMeeting[j]
            }
        }
        maxUserPerMeeting[i] += maxUserCount
    }
    println(maxUserPerMeeting.max())

}

data class Meeting(
    val startTime: Int,
    val endTime: Int,
    val users: Int,
)